crates/sui-core/src/authority_aggregator.rs

```diff
@@ -1182,24 +1182,11 @@ where
             .await
     }
 
-    /// Takes a transaction, brings all authorities up to date with the versions of the
-    /// objects needed, and then submits the transaction to make a certificate.
+    /// Submits the transaction to a quorum of validators to make a certificate.
     pub async fn process_transaction(
         &self,
         transaction: Transaction,
     ) -> Result<CertifiedTransaction, SuiError> {
-        // Find out which objects are required by this transaction and
-        // ensure they are synced on authorities.
-        let required_ids: Vec<ObjectID> = transaction
-            .data
-            .input_objects()?
-            .iter()
-            .map(|o| o.object_id())
-            .collect();
-
-        let (_active_objects, _deleted_objects) =
-            self.sync_all_given_objects(&required_ids).await?;
```

Comment:
Not sure how far you want to go with this. However, if you also remove `sync_account_state` (as suggested here https://github.com/MystenLabs/sui/issues/1045) you can also remove `sync_all_owned_objects` which is the only call site of `sync_all_given_objects`, and thus you can remove that too.